### PR TITLE
[MIG] account_cost_center: Migration to 15.0

### PR DESCRIPTION
**Pull request by smemon786**

### CODE REVIEW

**Comment by smemon786** [comment]
@aisopuro Please review it.

**Comment by kittiu** [comment]
Hello, thanks for contributing.
Please make sure you follow the guideline in order to migrate https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-15.0#technical-method-to-migrate-a-module-from-140-to-150-branch

**Comment by smemon786** [comment]
@kittiu + @aisopuro this module is ready to merge.

**Comment by smemon786** [comment]
> Hello, thanks for contributing. Please make sure you follow the guideline in order to migrate https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-15.0#technical-method-to-migrate-a-module-from-140-to-150-branch

@kittiu, this module is ready to merge, please have a look.

**Comment by aisopuro** [comment]
@smemon786 it seems this PR is only 1 commit, but I suspect it should be several, in order to maintain the original history of the module.

The link you've added is for migrating a module _that existed in 14.0_ to 15.0. But it looks like `account_cost_center` hasn't been in the repo since version 12.0.

I think you need to adapt the commands you linked to use branch `12.0` where they refer to `14.0`. That way you'll retain the proper git history of the module.

I basically think the first set of commands should be

```
$ git clone https://github.com/OCA/$REPO -b 15.0
$ cd $REPO
$ git checkout -b 15.0-mig-$MODULE origin/15.0
$ git format-patch --keep-subject --stdout origin/15.0..origin/12.0 -- $MODULE | git am -3 --keep
$ pre-commit run -a  # to run black, isort and prettier (ignore pylint errors at this stage)
$ git add -A
$ git commit -m "[IMP] $MODULE: black, isort, prettier"  --no-verify  # it is important to do all formatting in one commit the first time
```

**Comment by smemon786** [comment]
> @smemon786 it seems this PR is only 1 commit, but I suspect it should be several, in order to maintain the original history of the module.
> 
> The link you've added is for migrating a module _that existed in 14.0_ to 15.0. But it looks like `account_cost_center` hasn't been in the repo since version 12.0.
> 
> I think you need to adapt the commands you linked to use branch `12.0` where they refer to `14.0`. That way you'll retain the proper git history of the module.
> 
> I basically think the first set of commands should be
> 
> ```
> $ git clone https://github.com/OCA/$REPO -b 15.0
> $ cd $REPO
> $ git checkout -b 15.0-mig-$MODULE origin/15.0
> $ git format-patch --keep-subject --stdout origin/15.0..origin/12.0 -- $MODULE | git am -3 --keep
> $ pre-commit run -a  # to run black, isort and prettier (ignore pylint errors at this stage)
> $ git add -A
> $ git commit -m "[IMP] $MODULE: black, isort, prettier"  --no-verify  # it is important to do all formatting in one commit the first time
> ```

@aisopuro, please review now.

File: account_cost_center/models/account_invoice_report.py [hunk 1]
```diff
@@ -7,21 +7,10 @@
...
-            super(AccountInvoiceReport, self)._select()
-            + ", sub.cost_center_id as cost_center_id"
-        )
-
-    def _sub_select(self):
-        return (
-            super(AccountInvoiceReport, self)._sub_select()
-            + ", ail.cost_center_id as cost_center_id"
-        )
+        return super()._select() + ", line.cost_center_id as cost_center_id"
```
Comment (aisopuro) [inline account_cost_center/models/account_invoice_report.py]: This line is missed by coverage. Could we add a simple test to just `search_read` from `account.invoice.report`? I believe that should be enough to exercise this line.

File: account_cost_center/models/account_invoice_report.py [hunk 2]
```diff
@@ -7,21 +7,10 @@
...
 
     def _group_by(self):
-        return super(AccountInvoiceReport, self)._group_by() + ", ail.cost_center_id"
+        return super()._group_by() + ", line.cost_center_id"
```
Comment (aisopuro) [inline account_cost_center/models/account_invoice_report.py]: Same as above, I think just doing a `search_read` should cover this line.

**Review by aisopuro — CHANGES_REQUESTED** [review]

**Review by aisopuro — APPROVED** [review]

**Comment by anikeenko-viktor** [comment]
Could you please @kittiu check and merge it?

**Comment by kittiu** [comment]
@TheerayutEncoder can you help test?

**Comment by TheerayutEncoder** [comment]
OK, I will test this PR tomorrow.
For make sure it work properly in v15!
@kittiu

**Comment by TheerayutEncoder** [comment]
Hi!
No error occurred while I installed this module.
But I can not create **Cost Centers** menu because it does not have **Analytics accounting** in Configuration menu
I already installed **Invoicing** and **Analytic Accounting**.

or should I setting something first?
@aisopuro 
Thanks in advance

**Review by TheerayutEncoder — APPROVED** [review]
This module work on Odoo v15.0 [👍](https://emojipedia.org/thumbs-up/)

**Comment by anikeenko-viktor** [comment]
Hi @kittiu, could you please merge this one?

**Pull request merged**